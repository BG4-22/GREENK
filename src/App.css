.App {
    text-align: center;
}

#navbar {
    position: absolute;
    top: 0;
    left: 0;
}
