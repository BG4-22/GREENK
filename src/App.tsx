import { Navigate, Route, Routes } from 'react-router-dom';
import Layout from './layout/Layout';
import EnergyFlow from './pages/EnergyFlow';
import Game from './pages/Game';
import Home from './pages/Home';
import Statistics from './pages/Statistics';

/**
 * Main react component. Here the routes are made to map each url path and the components that is shown on each of those.
 */

function App() {
    return (
        <Routes>
            <Route path="/" element={<Layout />}>
                <Route index element={<Home />} />
                <Route path="statistikk" element={<Statistics />} />
                <Route path="spill" element={<Game />} />
                <Route path="energiflyt" element={<EnergyFlow />} />
                <Route path="*" element={<Navigate replace to={'/'} />} />
            </Route>
        </Routes>
    );
}

export default App;
