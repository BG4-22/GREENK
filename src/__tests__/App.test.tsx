import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../App';

describe('<App />', () => {
    const wrapper = shallow(<App />);
    it('Should render correctly', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper).toHaveLength(1);
    });
});
