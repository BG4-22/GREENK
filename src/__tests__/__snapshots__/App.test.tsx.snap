// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render correctly 1`] = `
<Routes>
  <Route
    element={<Layout />}
    path="/"
  >
    <Route
      element={<Home />}
      index={true}
    />
    <Route
      element={<Statistics />}
      path="statistikk"
    />
    <Route
      element={<Game />}
      path="spill"
    />
    <Route
      element={<EnergyFlow />}
      path="energiflyt"
    />
    <Route
      element={
        <Navigate
          replace={true}
          to="/"
        />
      }
      path="*"
    />
  </Route>
</Routes>
`;
